mongo/errors.go

```diff
@@ -285,7 +299,12 @@ func (e CommandError) Unwrap() error {
 
 // HasErrorCode returns true if the error has the specified code.
 func (e CommandError) HasErrorCode(code int) bool {
-	return int(e.Code) == code
+	return hasErrorCode(e, code)
```

Comment:
Re-implementing this in terms of `ErrorCodes` seems like an unnecessary change. The existing code is still valid and is more obvious.